modules/php/States/PlayCardTrait.php

```diff
@@ -26,98 +24,132 @@ public function st_playCard()
+        $freeRulesAvailable[] = [
+          "card_id" => $card_id,
+          "name" => $ruleCard->getName(),
+        ];
+      }
     }
 
-    $addInflation = Utils::getActiveInflation() ? 1 : 0;
-    // check bonus rules
-    $partyBonus =
-      Utils::getActivePartyBonus() && Utils::isPartyInPlay()
-        ? 1 + $addInflation
-        : 0;
-    if ($partyBonus > 0 && $withNotifications) {
-      RulePartyBonus::notifyActiveFor($player_id, false);
-    }
-    $richBonus =
-      Utils::getActiveRichBonus() && Utils::hasMostKeepers($player_id)
-        ? 1 + $addInflation
-        : 0;
-    if ($richBonus > 0 && $withNotifications) {
-      RuleRichBonus::notifyActiveFor($player_id);
-    }
+    return $freeRulesAvailable;
+  }
 
-    // Play All but 1 is also affected by Inflation and Bonus rules
-    if ($playRule < 0) {
-      $playRule -= $addInflation;
-      // if "Play All but ..." + bonus plays becomes >= 0, it actually becomes "Play All"
-      if ($playRule + $partyBonus + $richBonus >= 0) {
-        return PLAY_COUNT_ALL;
-      }
-      // else it stays "Play All but ..."
-      return $playRule + $partyBonus + $richBonus;
+  public function action_finishTurn()
+  {
+    $game = Utils::getGame();
+    // Check that this is the player's turn and that it is a "possible action" at this game state (see states.inc.php)
+    $game->checkAction("finishTurn");
+
+    $player_id = $game->getActivePlayerId();
+    if ($this->activePlayerMustPlayMoreCards($player_id)) {
+      Utils::throwInvalidUserAction(
+        fluxx::totranslate(
+          "You cannot finish your turn if you still need to play cards"
+        )
+      );
     }
-    // Normal Play Rule
-    else {
-      $playRule += $addInflation + $partyBonus + $richBonus;
+
+    $game->gamestate->nextstate("endOfTurn");
+  }
+
+  public function action_playFreeRule($card_id)
+  {
+    $game = Utils::getGame();
+
+    // Check that this is the player's turn and that it is a "possible action" at this game state (see states.inc.php)
+    $game->checkAction("playFreeRule");
+
+    $player_id = $game->getActivePlayerId();
+    $card = $game->cards->getCard($card_id);
+
+    if ($card["location"] != "rules") {
+      Utils::throwInvalidUserAction(
+        fluxx::totranslate("This is not an active Rule")
+      );      
     }
 
-    return $playRule;
+    $ruleCard = RuleCardFactory::getCard($card_id, $card["type_arg"]);
+
+    $game->notifyAllPlayers(
+      "freeRulePlayed",
+      clienttranslate('${player_name} uses free rule <b>${card_name}</b>'),
+      [
+        "i18n" => ["card_name"],
+        "player_name" => $game->getActivePlayerName(),
+        "player_id" => $player_id,
+        "card_name" => $ruleCard->getName(),
+      ]
+    );    
+
+    $stateTransition = $ruleCard->freePlayInPlayerTurn($player_id);
+    if ($stateTransition != null) {
+      // player must resolve something before continuing to play more cards
+      $game->gamestate->nextstate($stateTransition);
```

Comment:
Oh actually, that's how you implemented it. So you can simply return something different than `null` in those 2 cases.

Reply:
Yes, see other comment, to be checked if we can just transition via "endOfTurn" here, that would be better.

Reply:
So done